src/DynamoApplications/AssemblyConflictChecker.cs

```diff
@@ -0,0 +1,134 @@
+
+            foreach (var conflict in conflicts)
+            {
+                var ignored = ignoreList.Contains(conflict.Key);
+                Conflicts.Add(conflict.Value, ignored);
+            }
+        }
+
+        /// <summary>
+        /// Constructor for AssemblyConflictNotifier
+        /// </summary>
+        public AssemblyConflictChecker()
+        {
+            Enable = true;
+            AppDomain.CurrentDomain.AssemblyLoad += AssemblyLoad;
+        }
+
+        /// <summary>
+        /// Dispose function for AssemblyConflictNotifier
+        /// </summary>
+        public void Dispose()
+        {
+            AppDomain.CurrentDomain.AssemblyLoad -= AssemblyLoad;
+        }
+
+        /// <summary>
+        /// Logs any assembly conflicts found.
+        /// </summary>
+        public void LogConflicts(Models.DynamoModel DynamoModel)
+        {
+            foreach (var conflict in Conflicts)
+            {
+                bool isIgnored = conflict.Value;
+                var x = conflict.Key;
+
+                if (isIgnored)
+                {
+                    DynamoModel.Logger.LogInfo(GetType().ToString(), x.Message);
+                } else
+                {
+                    DynamoModel.Logger.LogNotification(DynamoModel.GetType().ToString(),
+                    x.GetType().ToString(), Resources.MismatchedAssemblyVersionShortMessage,
+                    x.Message);
+                }
+            }
+            Conflicts.Clear();
+        }
+
+        internal static Dictionary<string, Exception> GetVersionMismatchedReferencesInAppDomain(Assembly assembly)
+        {
+            // Get all assemblies that are currently loaded into the appdomain.
+            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies().ToList();
+            // Ignore some assemblies(Revit assemblies) that we know work and have changed their version number format or do not align
+            // with semantic versioning.
+
+            var loadedAssemblyNames = loadedAssemblies.Select(assem => assem.GetName()).ToList();
+
+            //build dict- ignore those with duplicate names.
+            var loadedAssemblyDict = loadedAssemblyNames.GroupBy(assm => assm.Name).ToDictionary(g => g.Key, g => g.First());
+
+            var output = new Dictionary<string, Exception>();
+            foreach (var currentReferencedAssembly in assembly.GetReferencedAssemblies().Concat(new AssemblyName[] { assembly.GetName() }))
+            {
+                if (loadedAssemblyDict.ContainsKey(currentReferencedAssembly.Name))
+                {
+                    //if the dll is already loaded, then check that our required version is not greater than the currently loaded one.
+                    var loadedAssembly = loadedAssemblyDict[currentReferencedAssembly.Name];
+                    if (currentReferencedAssembly.Version.Major > loadedAssembly.Version.Major)
```

Comment:
do you think this should also do a check for  versions less than the loaded major version?